nixos/modules/services/misc/chatwoot.nix

```diff
@@ -0,0 +1,155 @@
+    partOf = ["chatwoot.target"];
+
+    path = pkgs.chatwoot.passthru.extraPath;
+
+    environment = {
+      PORT = toString cfg.port;
+      RAILS_ENV = "production";
+      NODE_ENV = "production";
+      RAILS_LOG_TO_STDOUT = "true";
+      FRONTEND_URL = cfg.externalUrl;
+      HOME = "/run/chatwoot";
+
+      POSTGRES_DATABASE = "chatwoot";
+      POSTGRES_USERNAME = "chatwoot";
+      POSTGRES_HOST = "";
+
+      REDIS_URL = "redis://localhost:6379";
+    } // cfg.extraConfig;
+
+    serviceConfig = {
+      Type = "simple";
+      DynamicUser = true;
+      User = "chatwoot";
+
+      RuntimeDirectory = "chatwoot";
+      RuntimeDirectoryMode = 0775;
+      WorkingDirectory = pkgs.chatwoot;
+      LogDirectory = "chatwoot";
+      StateDirectory = "chatwoot";
+
+      Restart = "always";
+      RestartSec = 1;
+      TimeoutStopSec = 30;
+      KillMode = "mixed";
+      StandartInput = null;
+      SyslogIdentifier = "%p";
+    };
+  };
+in
+{
+  options.services.chatwoot = {
+    enable = mkEnableOption "chatwoot";
+
+    externalUrl = mkOption {
+      type = types.str;
+      description = "URL over which chatwoot will be accessed";
+    };
+
+    domain = mkOption {
+      type = types.str;
+      description = "Domain under which to run chatwoot";
+    };
+
+    port = mkOption {
+      description = "Port to listen at";
+      type = types.int;
+      default = 3000;
+    };
+
+    openFirewall = mkOption {
+      type = types.bool;
+      default = false;
+      description = "Open ports in the firewall for chatwoot.";
+    };
+
+    extraConfig = mkOption {
+      type = types.attrs;
+      description = ''
+        Extra config options
+
+        See https://github.com/chatwoot/chatwoot/blob/develop/.env.example
+
+        Example
+        <literal>{
+          ENABLE_ACCOUNT_SIGNUP = true;
+          MAILER_SENDER_EMAIL = "YourCompany &lt;no-reply@company.tld&gt;";
+        }</literal>
+        '';
+      default = {};
+    };
+
+    nginx = mkEnableOption "chatwoot nginx configuration";
+  };
+
+  config = mkIf (cfg.enable) {
+    services.chatwoot.domain = mkDefault "https://${cfg.domain}";
+
+    networking.firewall = mkIf cfg.openFirewall {
+      allowedTCPPorts = [ cfg.port ];
+    };
+
+    services.redis = {
+      enable = true;
+    };
+
+    services.postgresql = {
+      enable = true;
+      ensureUsers = [{
+        name = "chatwoot";
+        ensurePermissions = { "DATABASE chatwoot" = "ALL PRIVILEGES"; };
+      }];
+      ensureDatabases = [ "chatwoot" ];
+    };
+
+    services.nginx = mkIf (cfg.nginx) {
+      enable = true;
+
+      recommendedTlsSettings = true;
+      recommendedOptimisation = true;
+      recommendedGzipSettings = true;
+      recommendedProxySettings = true;
+
+      virtualHosts.${cfg.domain} = {
+        enableACME = true;
+
+        locations."/" = {
+          proxyPass = "http://localhost:${toString cfg.port}";
+          proxyWebsockets = true;
+        };
+      };
+    };
```

Comment:
I am not sure if we should enforce those settings on the end user.

Reply:
These are adapted from https://github.com/chatwoot/chatwoot/blob/develop/deployment/nginx_chatwoot.conf

It's optional for convenience and can be enabled with services.chatwoot.nginx